cranelift-codegen/src/isa/x86/unwind.rs

```diff
@@ -0,0 +1,525 @@
+//! Unwind information for x64 Windows.
+use super::registers::RU;
+use crate::ir::{Function, InstructionData, Opcode};
+use crate::isa::{CallConv, RegUnit, TargetIsa};
+use alloc::vec::Vec;
+
+// Maximum (inclusive) size of a "small" stack allocation
+const SMALL_ALLOC_MAX_SIZE: u32 = 128;
+// Maximum (inclusive) size of a "large" stack allocation that can represented in 16-bits
+const LARGE_ALLOC_16BIT_MAX_SIZE: u32 = 524280;
+
+struct MemoryWriter {
+    ptr: *mut u8,
+    offset: usize,
```

Comment:
Is there any reason you cant use a plain `Vec<u8>`? That would be a lot safer.

Reply:
Originally I was passing in a `MemoryCodeSink`, which was a little too "heavy" of an abstraction for what this code was doing, so I ripped off it's memory writing functionality.

I agree that this could just simply write to the `Vec` directly which would be safer.  I'll remove this in favor of that.